tests/Fixer/ControlStructure/NoUnneededNestedControlBlockFixerTest.php

```diff
@@ -0,0 +1,577 @@
+
+        echo 1;
+        echo 2;
+        echo 3;
+
+}
+',
+            '<?php
+if ($foo) {
+
+} else {
+    if ($bar && $c) {
+        echo 1;
+        echo 2;
+        echo 3;
+    }
+}
+',
+        ];
+
+        yield 'comment "else" case' => [
+            '<?php
+if ($foo) {
+
+} elseif (/* 2 */$bar) {
+    /* 1 *//* 3 */ /* 4 */
+        echo 1;
+
+}
+',
+            '<?php
+if ($foo) {
+
+} else {
+    /* 1 */ if (/* 2 */$bar)/* 3 */ { /* 4 */
+        echo 1;
+    }
+}
+',
+        ];
+
+        // precedence cases
+
+        yield '"if" case precedence "and"' => [
+            '<?php
+if (($foo and $c) && $bar) {
+
+
+
+}
+',
+            '<?php
+if ($foo and $c) {
+    if ($bar) {
+
+    }
+}
+',
+        ];
+
+        $precedenceCases = [
+            '$x && ($y %= $b)' => '$y %= $b',
+            '$x && ($y &= $b)' => '$y &= $b',
+            '$x && ($y **= $b)' => '$y **= $b',
+            '$x && ($y *= $b)' => '$y *= $b',
+            '$x && ($y += $b)' => '$y += $b',
+            '$x && ($y -= $b)' => '$y -= $b',
+            '$x && ($y .= $b)' => '$y .= $b',
+            '$x && ($y /= $b)' => '$y /= $b',
+            '$x && ($y <<= $b)' => '$y <<= $b',
+            '$x && ($y = $b)' => '$y = $b',
+            '$x && ($y >>= $b)' => '$y >>= $b',
+            '$x && ($y ? $a : $b)' => '$y ? $a : $b',
+            '$x && ($y ^= $b)' => '$y ^= $b',
+            '$x && ($y and $b)' => '$y and $b',
+            '$x && ($y or $b)' => '$y or $b',
+            '$x && ($y xor $b)' => '$y xor $b',
+            '$x && ($y |= $b)' => '$y |= $b',
+            '$x && ($y || $b)' => '$y || $b',
+        ];
+
+        foreach ($precedenceCases as $combined => $second) {
+            yield [
+                sprintf(
+                    '<?php
+function foo(){
+if (%s) {
+
+
+
+}
+}
+',
+                    $combined
+                ),
+                sprintf(
+                    '<?php
+function foo(){
+if ($x) {
+    if (%s) {
+
+    }
+}
+}
+',
+                    $second
+                ),
+            ];
+        }
+
+        yield 'already wrapped precedence' => [
+            '<?php
+if ($foo && $a && ($a xor $b)) {
+
+
+
+}
+',
+            '<?php
+if ($foo) {
+    if ($a && ($a xor $b)) {
+
+    }
+}
+',
+        ];
+
+        // multiple flat
+
+        $expected = str_repeat("if (\$foo && \$bar) {\n\n# 1\n\n}\n", 10);
+        $input = str_repeat("if (\$foo) {\n    if (\$bar) {\n# 1\n    }\n}\n", 10);
+
+        yield 'multiple flat "if" case' => [
+            "<?php\n".$expected,
+            "<?php\n".$input,
+        ];
+
+        // multiple nested
+
+        yield 'multiple "if" case' => [
+            '<?php
+if ($foo && $bar1 && $bar2 && $bar3) {
+
+
+
+
+
+
+
+}
+',
+            '<?php
+if ($foo) {
+    if ($bar1) {
+        if ($bar2) {
+            if ($bar3) {
+
+            }
+        }
+    }
+}
+',
+        ];
+
+        yield 'multiple "else" case' => [
+            '<?php
+if ($foo) {
+
+} else {
```

Comment:
The remaining `else` blocks are unneeded and should be removed as well.

Reply:
The remaining ones are not nested, so out of scope. We already have a fixer for the useless else and a priority check for it. So I think it is not needed to remove the else block here.

Reply:
> The remaining ones are not nested

I don't understand. The expected part of this test case does have `if`s inside `else`s, and [the test case at line 105](https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/4799/files#diff-357a383ebe0f37632dbc43479ea5661cR105) implies they should be replaced with `elseif`s.

The following:
```php
if ($a) {
} elseif ($b) {
} elseif ($c) {
}
```
is the same as
```php
if ($a) {
} else {
    if ($b) {
    } else {
        if ($c) {
        }
    }
}
```
right?

Also I don't think any other fixer would take care of that (`no_useless_else` would only remove `else` if the previous `if` has e.g. a `return` statement).

Reply:
> `no_useless_else` would only remove else if the previous if has e.g. a return statement

It shouldn't, so if you have an example that fails please share it :)
For example:
```
   1) test.php (no_useless_else)
      ---------- begin diff ----------
--- Original
+++ New
@@ @@
 <?php
 if (1) {
     echo 123;
-} else {
+}  
 
-}
+
```

I'll look into the other cases to see what can be improved :+1: